osh/cmd_eval.py

```diff
@@ -1451,6 +1511,19 @@ def _Dispatch(self, node, cmd_st):
 
                 status = 0
 
+            elif case(command_e.Func):
+                node = cast(command.Func, UP_node)
+
+                name = lexer.TokenVal(node.name)
+                if name in self.funcs and not self.exec_opts.redefine_proc():
+                    e_die(
+                        "Func %s was already defined (redefine_proc)" %
+                        node.name, node.name)
+
+                self.funcs[name] = Func(node, self.mem, self)
```

Comment:
minor thing (since  I have looked at GC pressure too much) -- probably should just pass in `name` to avoid calling lexer.TokenVal() twice

that will produce 2 separate strings when we only need 1